frontend/src/shared/helpers.tsx

```diff
@@ -577,25 +580,44 @@ export function prepareCohortDataForAgGrid(
 
   let newCohortsConnection = { ...cohortsConnection };
 
-  const selectedAll = JSON.stringify(filterModel) === "{}";
-  if (!selectedAll) {
+  if ("initialCohortDeliveryDate" in filterModel) {
+    const { dateFrom, dateTo } = filterModel.initialCohortDeliveryDate;
+    newCohorts = newCohorts.filter((cohort) => {
+      const deliveryDate = cohort.initialCohortDeliveryDate;
+      if (!deliveryDate) return false; // handles moment(undefined) returning today's date
+      return (
+        moment(deliveryDate).isSameOrAfter(dateFrom) &&
+        moment(deliveryDate).isSameOrBefore(dateTo)
+      );
+    });
+    newCohortsConnection.totalCount = newCohorts.length;
+  }
+
+  if ("billed" in filterModel) {
     const selectedValues = filterModel.billed.values;
     const selectedNone = selectedValues.length === 0;
-
     if (selectedNone) {
       newCohorts = [];
       newCohortsConnection.totalCount = 0;
     } else {
-      newCohorts = newCohorts.filter(
-        (cohort) => cohort.billed === selectedValues[0]
+      newCohorts = newCohorts.filter((cohort) =>
+        selectedValues.includes(cohort.billed)
       );
       newCohortsConnection.totalCount = newCohorts.length;
     }
   }
```

Comment:
Every time the user performs a filter, the entire Cohorts dataset is fetched, then this code filters the data based on the user's filter inputs, and finally a filtered data table is rendered to the user.